pkgs/by-name/so/sobjectizer/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  callPackage,
+  # attention:
+  #   static and shared are both required for tests building(buildTests)
+  withStatic ? true,
+  withShared ? true,
+  buildExamples ? false,
+  buildTests ? true,
+  buildAll ? false,
+}:
+
+# Derivation build example:
+#   $ nix-build -A sobjectizer
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sobjectizer";
+  version = "5.8.4";
+
+  src = fetchFromGitHub {
+    owner = "Stiffstream";
+    repo = "sobjectizer";
+    tag = "v.${finalAttrs.version}";
+    hash = "sha256-tIqWgd6TppHfqZk3XHzhG0t+Nn8BQCTP81UD7ls67UE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeDir = "../dev";
+
+  cmakeFlags = [
+    (lib.cmakeBool "SOBJECTIZER_BUILD_STATIC" withStatic)
+    (lib.cmakeBool "SOBJECTIZER_BUILD_SHARED" withShared)
+    (lib.cmakeBool "BUILD_EXAMPLES" buildExamples)
+    (lib.cmakeBool "BUILD_TESTS" buildTests)
+    (lib.cmakeBool "BUILD_ALL" buildAll)
+  ];
+
+  doCheck = buildTests || buildAll;
+
```

Comment:
In order to run the tests, you can add this here:
`passthru.tests = sobjectizer.override { withStatic = true; withShared = true; };`

`nix-build -A sobjectizer.passthru.tests`

Reply:
Sadly, this won't run automatically when building the package, so you'll need to run this command every time you update this package.

Reply:
> Sadly, this won't run automatically when building the package, so you'll need to run this command every time you update this package.

I see this like disadvantage. I'm fine with running tests only on changes.
IMHO, how it does now it's ok, clearly and understandably.

What I want is to _use_ this package. What I don't want is _unnecessary_ complication and __unnecessary_ debugging nix magic.

FYI: I currently use this package in my needs in initial version and this serve me good. I want to share this package with others but I don't want to go to adventure of "learning tricks" and co.

I liked your suggestions, thank you a lot!
- - -
If we done with this, please "Resolve conversation".

Thank you!

Reply:
```diff
-  # static and shared are both required for tests building.
-  doCheck = withStatic && withShared;
+  # The tests require the shared library thanks to the patch.
+  doCheck = withShared;
```

`passthru` won't be needed. Tests now run by default.